src/cli/rustup_mode.rs

```diff
@@ -774,6 +780,12 @@ fn check_updates(cfg: &Cfg) -> Result<()> {
 
 fn update(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
     let self_update = !m.is_present("no-self-update") && !self_update::NEVER_SELF_UPDATE;
+    let mut cfg = cfg.clone();
+    if let Some(p) = m.value_of("profile") {
+        let p = Profile::from_str(p)?;
+        cfg = cfg.with_profile(p);
+    }
+    let cfg = &cfg;
```

Comment:
I can see why you're doing this, but `Cfg` is meant to represent the filesystem configuration state more I think than the runtime state.  `toolchain.install_from_dist()` and friends (and yes it's a deep chain) would probably benefit from passing this profile through instead.